icu4c/APIChangeReport.md

```diff
@@ -5,265 +5,1477 @@
  License & terms of use: http://www.unicode.org/copyright.html
 -->
 
-# ICU4C API Comparison: ICU 68 (update #1: 68.2) with ICU 69
+# ICU4C API Comparison: ICU 69 with ICU 70 (m1)
 
 > _Note_ Markdown format of this document is new for ICU 65.
 
-- [Removed from ICU 68](#removed)
-- [Deprecated or Obsoleted in ICU 69](#deprecated)
-- [Changed in  ICU 69](#changed)
-- [Promoted to stable in ICU 69](#promoted)
-- [Added in ICU 69](#added)
-- [Other existing drafts in ICU 69](#other)
+- [Removed from ICU 69](#removed)
+- [Deprecated or Obsoleted in ICU 70](#deprecated)
+- [Changed in  ICU 70](#changed)
+- [Promoted to stable in ICU 70](#promoted)
+- [Added in ICU 70](#added)
+- [Other existing drafts in ICU 70](#other)
 - [Signature Simplifications](#simplifications)
 
 ## Removed
 
-Removed from ICU 68
+Removed from ICU 69
   
-| File | API | ICU 68 | ICU 69 |
+| File | API | ICU 69 | ICU 70 |
 |---|---|---|---|
-| listformatter.h | <tt>static</tt> ListFormatter* icu::ListFormatter::createInstance(const Locale&amp;, const char*, UErrorCode&amp;) |  Internal | (missing)
-| measunit.h | MeasureUnit icu::MeasureUnit::withSIPrefix(UMeasureSIPrefix, UErrorCode&amp;) const |  DraftICU 67 | (missing)
-| measunit.h | UMeasureSIPrefix icu::MeasureUnit::getSIPrefix(UErrorCode&amp;) const |  DraftICU 67 | (missing)
-| numberformatter.h | Usage&amp; icu::number::impl::Usage::operator=(Usage&amp;&amp;) |  Internal | (missing)
-| numberformatter.h | Usage&amp; icu::number::impl::Usage::operator=(const Usage&amp;) |  Internal | (missing)
-| numberformatter.h | bool icu::number::impl::Usage::isSet() const |  Internal | (missing)
-| numberformatter.h | icu::number::impl::Usage::Usage(Usage&amp;&amp;) |  Internal | (missing)
-| numberformatter.h | icu::number::impl::Usage::Usage(const Usage&amp;) |  Internal | (missing)
-| numberformatter.h | icu::number::impl::Usage::~Usage() |  Internal | (missing)
-| numberformatter.h | int16_t icu::number::impl::Usage::length() const |  Internal | (missing)
-| numberformatter.h | void icu::number::impl::Usage::set(StringPiece) |  Internal | (missing)
-| numberrangeformatter.h | UnicodeString icu::number::FormattedNumberRange::getFirstDecimal(UErrorCode&amp;) const |  DeprecatedICU 68 | (missing)
-| numberrangeformatter.h | UnicodeString icu::number::FormattedNumberRange::getSecondDecimal(UErrorCode&amp;) const |  DeprecatedICU 68 | (missing)
-| rbtz.h | void icu::RuleBasedTimeZone::getOffsetFromLocal(UDate, int32_t, int32_t, int32_t&amp;, int32_t&amp;, UErrorCode&amp;) const |  Internal | (missing)
-| simpletz.h | void icu::SimpleTimeZone::getOffsetFromLocal(UDate, int32_t, int32_t, int32_t&amp;, int32_t&amp;, UErrorCode&amp;) const |  Internal | (missing)
+| appendable.h | UBool icu::UnicodeStringAppendable::appendCodePoint(UChar32) |  StableICU 4.8 | (missing)
```

Comment:
The C++ API change report shows a misleadingly large number of changes from the addition of the `override` keyword that doesn't actually change the function signature.

@srl295 could you please point us to the code that simplifies function signatures for the report, so that irrelevant changes in annotations are not shown as a “change”? We should ignore both `virtual` and `override`.

Reply:
@markusicu https://github.com/unicode-org/icu/blob/24b3cf182e848e7ea23d7107a2289bbafff1fb8b/tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java#L543-L556

simplify here is in the sense of "create a canonical form for comparison".  You might want to make sure that override is at or near the end of the line, for example, so that some `u_overrideSomething()` does not get inadvertently transformed.

Reply:
is this the successor to U_OVERRIDE anyway? then maybe  ` "[ ]*(override|U_OVERRIDE)", ""`

Reply:
Thanks Steven!
Yes, we have recently just used `override` directly. `U_OVERRIDE` is only necessary for compilers that do not fully support C++11. We should be able to retire `U_OVERRIDE` unless the EBCDIC platforms come back with good-enough C++ support but still missing `override` -- and if we agree that that is “good enough”.

@sven-oly please change the tool as Steven provided here for your next C++ change report generation.

Reply:
Updated this line in the file, but the change report is still very large.

Line 511  "[ ]*(override|U_OVERRIDE)", "", // remove U_OVERRIDE

I also tried added a new line with just "override" but that also had no effect.

Reply:
Any other ideas to fix all these extra removes and adds?